src/sp/it/pl/audio/playlist/Playlist.java

```diff
@@ -369,7 +369,24 @@ public void updateItems() {
 		updateItems(this);
 	}
 
-	/******************************************************************************/
+/* ------------------------------------------------------------------------------------------------------------------ */
+
+	/** Plays given URI by searching for an existing PlaylistItem with that URI or adding it as a new Item */
+	public void playUri(URI uri) {
```

Comment:
These methods should take Item (which is based on the URI) as parameter. Optionally convenience methods for URI and other types could be introduced. But the URI is low level, Item is domain representation of the song. Which reminds me I should really rename the Item to Song...

I will accept it as is and take care of it in the future refactoring.